core/execution/execution.go

```diff
@@ -60,12 +60,13 @@ type Executor interface {
 	// - blockHeight: Height of block being created (must be > 0)
 	// - timestamp: Block creation time in UTC
 	// - prevStateRoot: Previous block's state root hash
+	// - metadata: Arbitrary data passed from the block manager (e.g., current header hash)
```

Comment:
nit: Arbitrary data sounds very weak. When used, are there any guarantees what data is available?